quarkus/tests/integration/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java

```diff
@@ -100,7 +103,34 @@ public void beforeEach(ExtensionContext context) throws Exception {
             onBeforeStartDistribution(context.getRequiredTestMethod().getAnnotation(BeforeStartDistribution.class));
 
             if (launch != null) {
-                result = dist.run(Arrays.asList(launch.value()));
+                List<String> argsFromLaunchAnnotation = new ArrayList<>(Arrays.asList(launch.value()));
+
+                WithLegacyStoreOnly withLegacyStoreOnly = context.getRequiredTestMethod().getAnnotation(WithLegacyStoreOnly.class);
+                if (withLegacyStoreOnly == null) {
+                    withLegacyStoreOnly = context.getTestClass()
```

Comment:
Could be good to verify also the inheritance of the test classes. I'm not sure if it's the case in the testsuite ATM, but it'd be good to resolve it now. The annotation presence for classes could be checked in the BeforeAll even with the inheritance and in BeforeEach check only the methods. WDYT?